daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/Typing.scala

```diff
@@ -269,6 +275,17 @@ private[validation] object Typing {
             throw EExpectedTemplatableType(env.ctx, tyConName)
         }
     }
+    mod.exceptions.foreach {
+      case (exnName, message) =>
+        val tyConName = TypeConName(pkgId, QualifiedName(mod.name, exnName))
+        val env = Env(languageVersion, world, ContextDefException(tyConName), Map.empty)
+        world.lookupDataType(env.ctx, tyConName) match {
+          case DDataType(_, ImmArray(), DataRecord(_)) =>
+            env.checkDefException(tyConName, message)
+          case _ =>
+            throw EExpectedExceptionableType(env.ctx, tyConName)
+        }
+    }
```

Comment:
Should also check that the exception is not a template.

Reply:
I will handle it a new PR.

Reply:
👍